### ISSUE TITLE
Should `needsEmailVerification` be considered an error?

### ISSUE THREAD

**Issue by plmercereau** [nhost/nhost#534]
Both `useSignUpEmailPassword` and `useSignInEmailPassword` have a `needsEmailVerification` property.
Moreover, `useSignInEmailPassword` also have a `needsMfaOtp` property.

However, when their value turns to be `true`, it is not flagged as an error, and `isSuccess` will return `true` when the handler has been called.

I think it makes sense when authenticating with `useSignUpEmailPassword`, as a user has been indeed added, but they still need to verify their email to authenticate. In that sense, it is not an error.

But it is less obvious with `useSignInEmailPassword`: the authentication actually failed if the email needs to be verified, and we should consider it an error.

To correct this behaviour will however introduce a breaking change, although minor.
Other hook using email verification: `useChangeEmail`.

@szilarddoro @elitan what do you think?

**Comment by szilarddoro** [comment]
It depends, because some apps are not restricting access to some of their services if the user has not verified themselves. Is it technically possible now to access features without verifying your email address?

**Comment by plmercereau** [comment]
yes, if email verification is not activated in the backend, then  authentication/registration succeeds straight ahead: user is returned, `isSuceess` is `true` and `needsEmailVerification` is `false`.
Note: `needsEmailVerification` will be `true`, but the email will still be flagged as unverified in the DB
